src/main/java/pro/sky/exever/examiner/ExaminerServiceImpl.java

```diff
@@ -0,0 +1,55 @@
+package pro.sky.exever.examiner;
+
+import org.springframework.stereotype.Service;
+import pro.sky.exever.examiner.exception.QuestionsOutOfBoundException;
+
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.concurrent.ThreadLocalRandom;
+
+@Service
+public class ExaminerServiceImpl implements ExaminerService {
+    private final JavaQuestionService javaQuestionService;
+    private final MathQuestionService mathQuestionService;
+
+    public ExaminerServiceImpl(JavaQuestionService javaQuestionService,
+                               MathQuestionService mathQuestionService) {
```

Comment:
```suggestion
    public ExaminerServiceImpl(QuestionService javaQuestionService,
                               QuestionService mathQuestionService) {
```
инъекцию надо делать по абстракции всегда (принцип L в SOLID)